url6a9f553762_Bresiu_socket_io_tgz-pJ8-MainJ8/src/com/github/nkzawa/socketio/client/IO.java

```diff
@@ -0,0 +1,91 @@
+package com.github.nkzawa.socketio.client;
+
+
+import com.github.nkzawa.socketio.parser.Parser;
+
+import javax.net.ssl.SSLContext;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+
+
+public class IO {
+
+    private static final Logger logger = Logger.getLogger(IO.class.getName());
+
+    private static final ConcurrentHashMap<String, Manager> managers = new ConcurrentHashMap<String, Manager>();
+
+    /**
+     * Protocol version.
+     */
+    public static int protocol = Parser.protocol;
+
+    public static void setDefaultSSLContext(SSLContext sslContext) {
+        Manager.defaultSSLContext = sslContext;
+    }
+
+    private IO() {}
+
+    public static Socket socket(String uri) throws URISyntaxException {
+        return socket(uri, null);
+    }
+
+    public static Socket socket(String uri, Options opts) throws URISyntaxException {
+        return socket(new URI(uri), opts);
+    }
+
+    public static Socket socket(URI uri) throws URISyntaxException {
+        return socket(uri, null);
+    }
+
+    /**
+     * Initializes a {@link Socket} from an existing {@link Manager} for multiplexing.
+     *
+     * @param uri uri to connect.
+     * @param opts options for socket.
+     * @return {@link Socket} instance.
+     * @throws URISyntaxException
+     */
+    public static Socket socket(URI uri, Options opts) throws URISyntaxException {
+        if (opts == null) {
+            opts = new Options();
+        }
+
+        URL parsed;
+        try {
+            parsed = Url.parse(uri);
+        } catch (MalformedURLException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)